version: "3.8"
services:
  postgres:
    image: postgres:13-alpine
    ports:
      - "127.0.0.1:7777:5432"
    environment:
      POSTGRES_DB: pydisapi
      POSTGRES_PASSWORD: pydisapi
      POSTGRES_USER: pydisapi
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U pydisapi" ]
      interval: 2s
      timeout: 1s
      retries: 5
    volumes:
      - ./postgres/init.sql:/docker-entrypoint-initdb.d/init.sql
  web:
    build:
      context: .
      dockerfile: Dockerfile
      target: development
    ports:
      - "127.0.0.1:8000:8000"
    depends_on:
      postgres:
        condition: service_healthy
    tty: true
    volumes:
      - .:/app:ro
      - ./alembic/versions:/app/alembic/versions
    environment:
      database_url: postgresql+asyncpg://pydisapi:pydisapi@postgres:5432/pydisapi
      auth_token: "my_token"
      debug: 1
